### PR TITLE
Automate ACK resource creation and add Makefile

### PR DESCRIPTION
**Pull request by akartsky**
Automate ACK resource creation and add Makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+install-python:
+	sudo apt install -q python3.8 -y 
+	sudo apt install -q python3-pip -y
+
+install-python-packages:
+	python3.8 -m pip install -r tests/e2e/requirements.txt
+
+install-all-prerequisites: install-awscli install-eksctl install-kubectl install-kustomize install-yq install-jq install-terraform install-helm install-python install-python-packages
+
+varify-cluster-variables:
```
Comment (mbaijal) [inline Makefile:58]: There is a typo - "verify"

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+connect-to-eks-cluster: varify-cluster-variables
+	aws eks update-kubeconfig --name $(CLUSTER_NAME) --region $(CLUSTER_REGION)
+
+port-forward:
+	$(eval IP_ADDRESS:=127.0.0.1)
+	$(eval PORT:=8080)
+	kubectl port-forward svc/istio-ingressgateway --address $(IP_ADDRESS) -n istio-system $(PORT):80
+
+bootstrap-ack: varify-cluster-variables connect-to-eks-cluster
```
Comment (rrrkharse) [inline Makefile:70]: varify-cluster-variables --> verify-cluster-variables

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+install-python-packages:
+	python3.8 -m pip install -r tests/e2e/requirements.txt
+
+install-all-prerequisites: install-awscli install-eksctl install-kubectl install-kustomize install-yq install-jq install-terraform install-helm install-python install-python-packages
+
+varify-cluster-variables:
+	test $(CLUSTER_NAME) || (echo Please export CLUSTER_NAME variable ; exit 1)
+	test $(CLUSTER_REGION) || (echo Please export CLUSTER_REGION variable ; exit 1)
+
+connect-to-eks-cluster: varify-cluster-variables
```
Comment (rrrkharse) [inline Makefile:62]: varify-cluster-variables --> verify-cluster-variables

File: tests/e2e/utils/ack_bootstrap/ack_deployment copy.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (rrrkharse) [inline tests/e2e/utils/ack_bootstrap/ack_deployment copy.py:1]: i think this file was added to this pr by accident

File: tests/e2e/utils/ack_bootstrap/ack_deployment.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                "Principal": {
+                    "Federated": f"arn:aws:iam::{account_id}:oidc-provider/{oidc_url}"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        f"{oidc_url}:aud": "sts.amazonaws.com",
+                        f"{oidc_url}:sub": [ 
+                            "system:serviceaccount:ack-system:ack-sagemaker-controller",
+                            "system:serviceaccount:ack-system:ack-applicationautoscaling-controller"
```
Comment (rrrkharse) [inline tests/e2e/utils/ack_bootstrap/ack_deployment.py:48]: nit: if we just add the sagemaker controller sa, which should be the only one we need, then we can use eksctl to create the sagemaker controller irsa

we can add this in a follow up pr

**Review by rrrkharse — CHANGES_REQUESTED** [review]

File: tests/e2e/utils/ack_bootstrap/ack_deployment.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+    print_banner("Reading Config")
+    cfg = load_yaml_file(file_path=config_file_path)
+
+    cluster_region = cfg["cluster"]["region"]
+    cluster_name = cfg["cluster"]["name"]
+    ACK_OIDC_ROLE_NAME = f"ack-controller-role-{cluster_name}"
+
+    # print_banner("Associate OIDC provider")
+    # associate_iam_oidc_provider(cluster_name=cluster_name, region=cluster_region)
```
Comment (rrrkharse) [inline tests/e2e/utils/ack_bootstrap/ack_deployment.py:130]: uncomment?

**Review by rrrkharse — CHANGES_REQUESTED** [review]

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+	kubectl version --client
+
+install-kustomize:
+	wget https://github.com/kubernetes-sigs/kustomize/releases/download/v3.2.0/kustomize_3.2.0_linux_amd64
+	chmod +x kustomize_3.2.0_linux_amd64
+	sudo mv kustomize_3.2.0_linux_amd64 /usr/local/bin/kustomize
+	kustomize version
+
+install-yq:
+	$(eval VERSION:=v4.26.1)
```
Comment (surajkota) [inline Makefile:30]: nit: YQ_VERSION

File: tests/e2e/utils/ack_bootstrap/ack_deployment.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    resp = iam_client.get_role(RoleName=role_name)
+    oidc_role_arn = resp["Role"]["Arn"]
+    return oidc_role_arn
+
+def get_account_id():
+    return boto3.client("sts").get_caller_identity().get("Account")
+
+def write_params(oidc_role_arn, region, file_path):
+    file_name = "params.env"
+    with open(f"{file_path}/{file_name}", "w") as file:
```
Comment (surajkota) [inline tests/e2e/utils/ack_bootstrap/ack_deployment.py:112]: see the configure_env_file https://github.com/awslabs/kubeflow-manifests/blob/main/tests/e2e/utils/load_balancer/setup_load_balancer.py#L113

**Review by surajkota — CHANGES_REQUESTED** [review]
looks good, few nits

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+port-forward:
+	$(eval IP_ADDRESS:=127.0.0.1)
+	$(eval PORT:=8080)
+	kubectl port-forward svc/istio-ingressgateway --address $(IP_ADDRESS) -n istio-system $(PORT):80
+
+bootstrap-ack: varify-cluster-variables connect-to-eks-cluster
+	yq e '.cluster.name=env(CLUSTER_NAME)' -i tests/e2e/utils/ack_bootstrap/config.yaml
+	yq e '.cluster.region=env(CLUSTER_REGION)' -i tests/e2e/utils/ack_bootstrap/config.yaml
+	cd tests/e2e && PYTHONPATH=.. python3.8 utils/ack_bootstrap/ack_deployment.py
```
[the remark this answers is not in the document]
Comment (akartsky) [reply Makefile:89]: cd is not retained 

every command runs from the root location

File: tests/e2e/utils/ack_bootstrap/ack_deployment.py (hunk 5 above)
Comment (akartsky) [reply tests/e2e/utils/ack_bootstrap/ack_deployment.py:49]: yep good suggestion
will create a common function which uses eksctl in a follow up pr

**Review by rrrkharse — APPROVED** [review]
leaving approval from my side

**Review by mbaijal — APPROVED** [review]

**Review by surajkota — APPROVED** [review]
approving for now

**Pull request merged**